core/gallery/gallery.go

```diff
@@ -228,3 +228,29 @@ func DeleteModelFromSystem(basePath string, name string, additionalFiles []strin
 
 	return err
 }
+
+// This is ***NEVER*** going to be perfect or finished.
+// This is a BEST EFFORT function to surface known-vulnerable models to users.
+func SafetyScanGalleryModels(galleries []config.Gallery, basePath string) error {
```

Comment:
This doesn't feel a good place for these functions, these are really huggingface-scoped ones. Maybe better to move those as `pkg/huggingface` ?

Reply:
The reason I didn't do this in the first place is that I wanted to keep the `core/config.Gallery` reference _out_ of the pkg layer. The part that would be safe to move to `pkg/huggingface` without risking any circular reference mess would be the part I already abstracted out to `pkg/downloader/uri.go`

I could pull `HuggingFaceScan` and `HuggingFaceScanResult` to a new package, and rename the call in `downloader` to not be HF-specific, adding an extra function call but making the organization a lot cleaner. I'll see how I like the look of that I guess.

Reply:
I played around with this and honestly, it seems messier to split it up than leaving that function within `pkg/downloader`. Without having any idea of what a non-HF scanner would expose to us... it's hard to split up cleanly. Not to say impossible - I am just not sure it's _better_ so I'll push up a changeset without it for review first.